__tests__/cache.test.ts

```diff
@@ -54,17 +55,48 @@ describe('InMemoryCache', () => {
         user: { name: 'Test' }
       }
     });
+
     jest.advanceTimersByTime(799);
     expect(Object.keys(cache.cache).length).toBe(1);
 
     jest.advanceTimersByTime(1);
     expect(Object.keys(cache.cache).length).toBe(0);
   });
 
+  it('strips everything except the refresh token when expiry has been reached', () => {
+    cache.save({
+      client_id: 'test-client',
+      audience: 'the_audience',
+      scope: 'the_scope',
+      id_token: 'idtoken',
+      access_token: 'accesstoken',
+      refresh_token: 'refreshtoken',
+      expires_in: 1,
+      decodedToken: {
+        claims: {
+          __raw: 'idtoken',
+          name: 'Test',
+          exp: new Date().getTime() / 1000 + 2
+        },
+        user: { name: 'Test' }
+      }
+    });
+
+    jest.advanceTimersByTime(799);
+    expect(Object.keys(cache.cache).length).toBe(1);
+
+    jest.advanceTimersByTime(1);
```

Comment:
This is a preexisting test from before this work went in, but essentially it tests that the cache is in one state right up to the expiry time, then as soon as the time advances onto the expiry time the cache should be in another state (i.e. it's been removed by a `setTimeout`).

Here, 800ms is used because Luis had a calculation in that defines the expiry time as `Math.min(expiresIn, expTime) * 1000 * 0.8`, and the `expiresIn` for this test is defined as 1000ms.

Reply:
Do you think something of the above, simplified of course, can be put as a comment to illustrate this testing intention? So future _me_ follows it

Reply:
Done!